2025-01-30 22:13:32,479 - __main__ - INFO - Skipping fragment creation: False
2025-01-30 22:13:32,715 - __main__ - DEBUG - Starting fragment retrieval process.
2025-01-30 22:13:33,175 - __main__ - DEBUG - Retrieved fragments: [{'content': '{"method": "POST", "parameters": [], "path": "/customers/subscribers", "requestBody": {"content": {"application/json": {"schema": {"properties": {"date_created": {"description": "The date on which the subscriber was created.\\n", "format": "data-time", "type": "string"}, "date_modified": {"description": "The date on which the subscriber was modified.\\n", "format": "data-time", "type": "string"}, "email": {"description": "The email of the subscriber. Must be unique.\\n", "type": "string"}, "first_name": {"description": "The first name of the subscriber.\\n", "type": "string"}, "id": {"description": "The unique numeric ID of the subscriber; increments sequentially.\\n", "type": "integer"}, "last_name": {"description": "The last name of the subscriber.\\n", "type": "string"}, "order_id": {"description": "The ID of the source order, if source was an order.\\n", "type": "integer"}, "source": {"description": "The source of the subscriber. Values are: `storefront`, `order`, or `custom`.\\n", "type": "string"}}, "type": "object"}}}, "description": "`Subscriber` object\\n", "required": true}, "responses": {"422": {"content": {"application/json": {"schema": {"allOf": [{"$ref": "#/components/schemas/BaseError"}, {"properties": {"errors": {"$ref": "#/components/schemas/DetailedErrors"}}, "type": "object"}]}}}, "description": "The `Subscriber` was not valid. This is the result of missing required fields, or of invalid data. See the response for more details.\\n"}}}', 'metadata': {'filename': '0157_customers_subscribers_post_422.json'}}, {'content': '{"method": "POST", "parameters": [], "path": "/customers/subscribers", "requestBody": {"content": {"application/json": {"schema": {"properties": {"date_created": {"description": "The date on which the subscriber was created.\\n", "format": "data-time", "type": "string"}, "date_modified": {"description": "The date on which the subscriber was modified.\\n", "format": "data-time", "type": "string"}, "email": {"description": "The email of the subscriber. Must be unique.\\n", "type": "string"}, "first_name": {"description": "The first name of the subscriber.\\n", "type": "string"}, "id": {"description": "The unique numeric ID of the subscriber; increments sequentially.\\n", "type": "integer"}, "last_name": {"description": "The last name of the subscriber.\\n", "type": "string"}, "order_id": {"description": "The ID of the source order, if source was an order.\\n", "type": "integer"}, "source": {"description": "The source of the subscriber. Values are: `storefront`, `order`, or `custom`.\\n", "type": "string"}}, "type": "object"}}}, "description": "`Subscriber` object\\n", "required": true}, "responses": {"200": {"content": {"application/json": {"schema": {"description": "Response payload for the Bigcommerce API.\\n", "properties": {"data": {"$ref": "#/components/schemas/Subscriber"}, "meta": {"$ref": "#/components/schemas/Meta"}}, "type": "object"}}}, "description": "A `Subscriber` object.\\n"}}}', 'metadata': {'filename': '0155_customers_subscribers_post_200.json'}}, {'content': '{"method": "POST", "parameters": [], "path": "/customers/subscribers", "requestBody": {"content": {"application/json": {"schema": {"properties": {"date_created": {"description": "The date on which the subscriber was created.\\n", "format": "data-time", "type": "string"}, "date_modified": {"description": "The date on which the subscriber was modified.\\n", "format": "data-time", "type": "string"}, "email": {"description": "The email of the subscriber. Must be unique.\\n", "type": "string"}, "first_name": {"description": "The first name of the subscriber.\\n", "type": "string"}, "id": {"description": "The unique numeric ID of the subscriber; increments sequentially.\\n", "type": "integer"}, "last_name": {"description": "The last name of the subscriber.\\n", "type": "string"}, "order_id": {"description": "The ID of the source order, if source was an order.\\n", "type": "integer"}, "source": {"description": "The source of the subscriber. Values are: `storefront`, `order`, or `custom`.\\n", "type": "string"}}, "type": "object"}}}, "description": "`Subscriber` object\\n", "required": true}, "responses": {"409": {"content": {"application/json": {"schema": {"allOf": [{"$ref": "#/components/schemas/BaseError"}, {"properties": {"errors": {"$ref": "#/components/schemas/DetailedErrors"}}, "type": "object"}]}}}, "description": "The `Subscriber` was in conflict with another subscriber. This is the result of duplicate unique values, such as `email`.\\n"}}}', 'metadata': {'filename': '0156_customers_subscribers_post_409.json'}}], form collection oas_bigecommerce_fragments
2025-01-30 22:13:33,177 - karate_llm_test_gen - INFO - Generating Karate test for prompt: 
Given the following API endpoint information:

  ...
2025-01-30 22:13:33,179 - karate_llm_test_gen - INFO - Getting relevant examples for: 
Given the following API endpoint information:

  ...
2025-01-30 22:13:36,812 - karate_llm_test_gen - INFO - Found 3 relevant examples.
2025-01-30 22:13:36,812 - karate_llm_test_gen - INFO - Sending request to http://192.168.1.108:4893/v1/chat/completions
2025-01-30 22:13:36,814 - karate_llm_test_gen - ERROR - Error: An error occurred while generating the Karate test: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
2025-01-30 22:13:36,814 - __main__ - INFO - Generated Karate test for /customers/subscribers POST at output/karate_feature_files/ID_001_krtestoas_customers_subscribers_POST.feature
2025-01-30 22:13:36,815 - karate_compatibility_verifier - INFO - Original content:
2025-01-30 22:13:36,815 - karate_compatibility_verifier - INFO - An error occurred while generating the Karate test: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
2025-01-30 22:13:36,838 - karate_compatibility_verifier - ERROR - An error occurred while calling the LLM: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
2025-01-30 22:13:36,838 - karate_compatibility_verifier - ERROR - 
Compatibility check failed. Using original content for syntax verification.
2025-01-30 22:13:36,842 - karate_compatibility_verifier - ERROR - An error occurred while calling the LLM: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
2025-01-30 22:13:36,842 - karate_compatibility_verifier - ERROR - 
Syntax verification failed. No changes were made.
2025-01-30 22:13:36,843 - __main__ - WARNING - Compatibility issues found in output/karate_feature_files/ID_001_krtestoas_customers_subscribers_POST.feature: An error occurred while generating the Karate test: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
2025-01-30 22:13:36,844 - karate_llm_test_gen - INFO - Generating Karate test for prompt: 
Given the following API endpoint information:

  ...
2025-01-30 22:13:36,846 - karate_llm_test_gen - INFO - Getting relevant examples for: 
Given the following API endpoint information:

  ...
2025-01-30 22:13:36,946 - karate_llm_test_gen - INFO - Found 3 relevant examples.
2025-01-30 22:13:36,946 - karate_llm_test_gen - INFO - Sending request to http://192.168.1.108:4893/v1/chat/completions
2025-01-30 22:13:36,948 - karate_llm_test_gen - ERROR - Error: An error occurred while generating the Karate test: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
2025-01-30 22:13:36,949 - __main__ - INFO - Generated Karate test for /customers/subscribers POST at output/karate_feature_files/ID_002_krtestoas_customers_subscribers_POST.feature
2025-01-30 22:13:36,949 - karate_compatibility_verifier - INFO - Original content:
2025-01-30 22:13:36,950 - karate_compatibility_verifier - INFO - An error occurred while generating the Karate test: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
2025-01-30 22:13:36,981 - karate_compatibility_verifier - ERROR - An error occurred while calling the LLM: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
2025-01-30 22:13:36,981 - karate_compatibility_verifier - ERROR - 
Compatibility check failed. Using original content for syntax verification.
2025-01-30 22:13:36,984 - karate_compatibility_verifier - ERROR - An error occurred while calling the LLM: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
2025-01-30 22:13:36,984 - karate_compatibility_verifier - ERROR - 
Syntax verification failed. No changes were made.
2025-01-30 22:13:36,984 - __main__ - WARNING - Compatibility issues found in output/karate_feature_files/ID_002_krtestoas_customers_subscribers_POST.feature: An error occurred while generating the Karate test: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
2025-01-30 22:13:36,985 - karate_llm_test_gen - INFO - Generating Karate test for prompt: 
Given the following API endpoint information:

  ...
2025-01-30 22:13:36,985 - karate_llm_test_gen - INFO - Getting relevant examples for: 
Given the following API endpoint information:

  ...
2025-01-30 22:13:37,072 - karate_llm_test_gen - INFO - Found 3 relevant examples.
2025-01-30 22:13:37,073 - karate_llm_test_gen - INFO - Sending request to http://192.168.1.108:4893/v1/chat/completions
2025-01-30 22:13:37,075 - karate_llm_test_gen - ERROR - Error: An error occurred while generating the Karate test: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
2025-01-30 22:13:37,076 - __main__ - INFO - Generated Karate test for /customers/subscribers POST at output/karate_feature_files/ID_003_krtestoas_customers_subscribers_POST.feature
2025-01-30 22:13:37,077 - karate_compatibility_verifier - INFO - Original content:
2025-01-30 22:13:37,077 - karate_compatibility_verifier - INFO - An error occurred while generating the Karate test: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
2025-01-30 22:13:37,101 - karate_compatibility_verifier - ERROR - An error occurred while calling the LLM: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
2025-01-30 22:13:37,101 - karate_compatibility_verifier - ERROR - 
Compatibility check failed. Using original content for syntax verification.
2025-01-30 22:13:37,103 - karate_compatibility_verifier - ERROR - An error occurred while calling the LLM: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
2025-01-30 22:13:37,103 - karate_compatibility_verifier - ERROR - 
Syntax verification failed. No changes were made.
2025-01-30 22:13:37,103 - __main__ - WARNING - Compatibility issues found in output/karate_feature_files/ID_003_krtestoas_customers_subscribers_POST.feature: An error occurred while generating the Karate test: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
2025-01-30 22:13:37,103 - __main__ - INFO - Moving generated Karate feature files...
2025-01-30 22:13:37,154 - __main__ - INFO - Creating the Karate Test Framework with generated feature files
2025-01-30 22:13:40,184 - __main__ - INFO - Skipping karate test run.
